spec/scry/client_spec.cr

```diff
@@ -5,13 +5,16 @@ module Scry
     describe "#send_message" do
       it "sends a valid NotificationMessage to the Client io" do
         build_failure = BuildFailure.from_json(BUILD_ERROR_EXAMPLE)
-        diagnostic = Protocol::Diagnostic.new(build_failure)
+        diagnostic = LSP::Protocol::Diagnostic.new(
```

Comment:
Since a build_failure is used at least twice here, and requires passing a lot of args, I'd suggest adding a helper method in `BuildFailure`, like `#to_lsp_diagnostic` to build the `LSP::Protocol::Diagnostic` instance